kebechet/config.py

```diff
@@ -46,36 +48,57 @@ def from_file(self, config_path: str):
                 content = config_file.read()
 
         try:
-            self._repositories = yaml.safe_load(content).pop('repositories') or []
+            self._repositories = yaml.safe_load(content).pop("repositories") or []
         except Exception as exc:
-            raise ConfigurationError("Failed to parse configuration file: {str(exc)}") from exc
+            raise ConfigurationError(
+                "Failed to parse configuration file: {str(exc)}"
+            ) from exc
+
+    def download_conf_from_url(self, url: str, service: str):
+        service = Service(service=service, url=url)
+        tempfile = service.download_kebechet_config()
+        return tempfile
+
+    def run_url(self, url: str, service: str):
+        temp_file = self.download_conf_from_url(url, service)
```

Comment:
Clean up after everything is done:

```python
        temp_file = self.download_conf_from_url(url, service)
        try:
            _LOGGER.info("Filename = %s", temp_file.name)
            _LOGGER.info(temp_file.read())
            self.run(temp_file.name)
        finally:
            os.remove(temp_file.name)
```

BTW from design POV, it would be better to have this logic explicit in the methods rather than relying on calllee to clean stuff (that might not happen).